libraries/azurerm_resource.rb

```diff
@@ -34,3 +30,11 @@ def credentials
     end
   end
 end
+
+class AzurermPluralResource < AzurermResource; end
+
+class AzurermSingularResource < AzurermResource
+  def exists?
```

Comment:
@clintoncwolfe can you recommend if we should do `exist` or `exists`?

Reply:
@dmccown `exists?` is preferred, and I assume `exist?` is a fallback. https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md#some-things-are-defined-for-you

Reply:
FilterTable, used for plural resources, uses the plural inflection, `exist?` 

Singular resources should use the singular inflection, `exists?`.

That's my take.